librz/bin/bobj_process.c

```diff
@@ -166,6 +166,13 @@ RZ_API bool rz_bin_object_process_plugin_data(RZ_NONNULL RzBinFile *bf, RZ_NONNU
 			o->info->lang = "go";
 			o->lang = RZ_BIN_LANGUAGE_GO;
 		}
+	} else if (o->info && RZ_STR_ISNOTEMPTY(o->info->compiler)) {
+		char *go_compiler = rz_bin_file_golang_compiler(bf);
+		if (go_compiler) {
+			o->info->lang = "go";
+			o->lang = RZ_BIN_LANGUAGE_GO;
+			o->info->compiler = rz_str_newf("%s %s", go_compiler, o->info->compiler);
+		}
```

Comment:
Please apply the following changes from line `166`

```c
	if (o->info) {
		char *go_compiler = rz_bin_file_golang_compiler(bf);
		if (go_compiler) {
			o->info->lang = "go";
			o->lang = RZ_BIN_LANGUAGE_GO;
		}
		if (RZ_STR_ISNOTEMPTY(o->info->compiler)) {
			char *merge = rz_str_newf("%s %s", go_compiler, o->info->compiler);
			free(o->info->compiler);
			free(go_compiler);
			o->info->compiler = merge;
		} else {
			free(o->info->compiler); // always free since it might be an invalid string but allocated.
			o->info->compiler = go_compiler;
		}
	}
```